src/node_modules/app/components/elementary/ElasticSearchField.svelte

```diff
@@ -166,54 +150,61 @@
 		dispatch('change', event.detail);
 	}
 
-	function getEditor (targetDataType, targetTypeObject) {
-		const baseType = extractBaseType(targetTypeObject);
-		if (targetDataType in typeEditors) {
-			return typeEditors[targetDataType]
-		} else if (baseType && isStringLiteralUnion(baseType)) {
-			const unionEditor = {
-				component: RadioList,
-				props: {
-					dataType: targetDataType,
-					list: targetDataType.split('|').map(i => i.trim().replaceAll('"',''))
-				}
-			};
-			return unionEditor;
-		} else if (isArrayString(targetDataType)) {
-			let subType;
-			if (isArrayString(baseType)) {
-				subType = baseType.substr(0, baseType.length - 2);
-			} else if (isArrayObject(baseType)) {
-				subType = baseType.__array;
-			}
-			const subDataType = targetDataType.substr(0, targetDataType.length - 2);
-			const arrayEditor = {
-				component: ArrayEditor,
-				props: {
-					fieldEditor: getEditor(subDataType, subType)
-				}
-			}
-			return arrayEditor;
-		} else if (equalsJson(targetDataType) || hasRecord(baseType)) {
-			return {
-				component: SimpleField,
-				props: {
-					type: 'json',
-					dataType: targetDataType
-				}
-			}
-		}
-		return  {
-			component: ObjectEditor,
-			props: {
-				type: 'json',
-				dataType: targetDataType,
-				typeObject: baseType
-			}
-		};
-	}
-
-	$: editor = getEditor(dataType, typeObject);
+	const makeTrimEndChars = count => string =>
+		string.substr(0, string.length - count);
+	const isOf = obj => key => key in obj;
+	const getFrom = obj => key => obj[key];
```

Comment:
these should be moved to `src/node_modules/app/utils/generic.js` as:

```suggestion
	const isKeyOf = _.curry(_.has)
	const getKeyOf = _.curry(_.getIn)
```